src/go/rpk/pkg/cli/registry/mode/set.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package mode
+
+import (
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/schemaregistry"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/sr"
+)
+
+var supportedModes = []string{"READONLY", "READWRITE"}
+
+func setCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var modeFlag string
+	cmd := &cobra.Command{
+		Use:   "set [SUBJECT...]",
+		Short: "Set schema registry mode",
+		Long: `Set schema registry mode
+
+The global schema registry mode can be set by passing an empty subject.`,
+		Example: `
+Set the global schema registry mode to READONLY
+  rpk registry mode set --mode READONLY
+
+Set the schema registry mode to READWRITE in subjects foo and bar
+  rpk registry mode set foo bar --mode READWRITE
+`,
```

Comment:
I would modify this text to something like this.  Feel free to not implement if it doesn't make sense. 
```suggestion
If you don't pass a subject, this command modifies the global schema registry mode. 

Acceptable mode values: 

- READONLY
- READWRITE

Examples: 

Set the schema registry mode to READWRITE in subjects foo and bar:
  rpk registry mode set foo bar --mode READWRITE
  
Set the global schema registry mode to READONLY:
  rpk registry mode set --mode READONLY
`,
```